src/System.ComponentModel.TypeConverter/tests/Design/DesignerVerbTests.cs

```diff
@@ -2,56 +2,153 @@
+        [InlineData("", "")]
+        [InlineData("Description", "Description")]
+        public void Description_GetWithProperties_ReturnsExpected(string value, string expected)
         {
-            var verb = new DesignerVerb("Text", new EventHandler(EventHandler)) { Description = value };
-            Assert.Equal(value ?? string.Empty, verb.Description);
+            var verb = new DesignerVerb("Text", new EventHandler(EventHandler));
+            verb.Properties["Description"] = value;
+            Assert.Equal(expected, verb.Description);
+        }
+
+        [Fact]
+        public void Description_GetWithPropertiesInvalidType_ThrowsInvalidCastException()
+        {
+            var verb = new DesignerVerb("Text", new EventHandler(EventHandler));
+            verb.Properties["Description"] = new object();
+            Assert.Throws<InvalidCastException>(() => verb.Description);
+        }
+
+        [Fact]
+        public void Description_GetWithNullProperties_ThrowsNullReferenceException()
+        {
+            var verb = new NullPropertiesAfterConstructionDesignerVerb("Text", new EventHandler(EventHandler));
+            Assert.Throws<NullReferenceException>(() => verb.Description = "value");
+        }
+
+        [Theory]
+        [InlineData(null, "")]
+        [InlineData("", "")]
+        [InlineData("Description", "Description")]
+        public void Description_Set_GetReturnsExpected(string value, string expected)
+        {
+            var verb = new DesignerVerb("Text", new EventHandler(EventHandler))
+            {
+                Description = value
+            };
+            Assert.Equal(expected, verb.Description);
+            Assert.Equal(value, verb.Properties["Description"]);
+
+            // Set same.
+            verb.Description = value;
+            Assert.Equal(expected, verb.Description);
+            Assert.Equal(value, verb.Properties["Description"]);
+        }
+
+        [Fact]
+        public void Description_SetWithNullProperties_ThrowsNullReferenceException()
+        {
+            var verb = new NullPropertiesAfterConstructionDesignerVerb("Text", new EventHandler(EventHandler));
+            Assert.Throws<NullReferenceException>(() => verb.Description);
+        }
+
+        [Theory]
+        [InlineData(null, "")]
+        [InlineData("", "")]
+        [InlineData("Text", "Text")]
+        public void Text_GetWithProperties_ReturnsExpected(string value, string expected)
+        {
+            var verb = new DesignerVerb("Text", new EventHandler(EventHandler));
+            verb.Properties["Text"] = value;
+            Assert.Equal(expected, verb.Text);
+        }
+
+        [Fact]
+        public void Text_GetWithPropertiesInvalidType_ThrowsInvalidCastException()
+        {
+            var verb = new DesignerVerb("Text", new EventHandler(EventHandler));
+            verb.Properties["Text"] = new object();
+            Assert.Throws<InvalidCastException>(() => verb.Text);
```

Comment:
Ditto: should we fix this invalid cast exception or are we happy with existing behavior?